lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -669,8 +669,8 @@ OpFoldResult ExpandDimsOp::fold(FoldAdaptor adaptor) {
 
 //-- ReshapeOp --
 template <typename OpType>
-LogicalResult canonicalizeViewOrBroadcast(OpType op,
```

Comment:
When doing reshape(reshape), we are biasing towards the parent reshape. We should set `allow-reorder` in the new op if one of them has it tho.

Reply:
Okay, I'm now propagating allow_reorder from either reshape op. PTAL